pytorch_lightning/core/lightning.py

```diff
@@ -1729,6 +1730,38 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
 
         torch.onnx.export(self, input_data, file_path, **kwargs)
 
+    def to_torchscript(self) -> Union[ScriptModule, Dict[str, ScriptModule]]:
+        """
+        By default compiles the whole model to a :class:`~torch.jit.ScriptModule`.
+        If you would like to customize the modules that are scripted or you want to use tracing
+        you should override this method. In case you want to return multiple modules, we
+        recommend using dictionary.
+
+        Note:
+            - Requires the implementation of the :meth:`LightningModule.forward` method.
+            - It is recommended that you install the latest supported version of PyTorch to use this feature without limitations.
+
+
+        Example:
+            >>> class SimpleModel(LightningModule):
+            ...     def __init__(self):
+            ...         super().__init__()
+            ...         self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+            ...
+            ...     def forward(self, x):
+            ...         return torch.relu(self.l1(x.view(x.size(0), -1)))
+            ...
+            >>> model = SimpleModel()
+            >>> torch.jit.save(model.to_torchscript(), "model.pt")  # doctest: +SKIP
+            >>> os.path.isfile("model.pt")  # doctest: +SKIP
+            True
+        """
+        mode = self.training
+        with torch.no_grad():
+            scripted_module = torch.jit.script(self.eval())
```

Comment:
I think it should be explicit stated that the scripted model is saved in evaluation mode.